src/iqm/qiskit_iqm/fake_backends/iqm_fake_backend.py

```diff
@@ -292,11 +303,26 @@ def run(self, run_input: Union[QuantumCircuit, list[QuantumCircuit]], **options)
         Raises:
             ValueError: If empty list of circuits is provided.
         """
-        circuits = [run_input] if isinstance(run_input, QuantumCircuit) else run_input
+        circuits_aux = [run_input] if isinstance(run_input, (QuantumCircuit, IQMCircuit)) else run_input
 
-        if len(circuits) == 0:
+        if len(circuits_aux) == 0:
             raise ValueError("Empty list of circuits submitted for execution.")
 
+        class moves_to_swaps(TransformationPass):
```

Comment:
The above code example is not valid for Deneb. Execute calls "transpile" followed by "run", but "transpile" is not defined for Deneb because of the Move. You need to use "transpile_to_iqm" followed by run to have the correct checks and balances.

Reply:
It works perfectly fine for a valid cricuit, however it would  check my code during compilation on server side.

```
    no_qubits = 6 
    comp_r = QuantumRegister(1, 'comp_r') # Computational resonator
    q = QuantumRegister(no_qubits, 'q') # Qubits 
    c = ClassicalRegister(no_qubits, 'c') # Classical register, used for readout
    qc = IQMCircuit(comp_r, q, c)

    qc.h(1)
    qc.x(2)
    qc.move(1,0)
    qc.cz(2,0)
    qc.move(1,0)
    qc.measure(q,c)

    job = execute(qc, backend, shots=1000)

    res=job.result()
    counts=res.get_counts()
    print(counts)
```